server/trillian_log_server.go

```diff
@@ -157,6 +181,20 @@ func (t *TrillianLogServer) commitAndLog(tx storage.LogTX, op string) error {
 	return err
 }
 
+func protoToLeaf(proto *trillian.LeafProto) trillian.LogLeaf {
+	return trillian.LogLeaf{Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
+}
+
+func protosToLeaves(protos []*trillian.LeafProto) []trillian.LogLeaf {
+	leaves := make([]trillian.LogLeaf, 0, len(protos))
+
+	for _, proto := range protos {
```

Comment:
In this case it doesn't matter as all elements are returned but if some were filtered out you'd have a slice with len(x) but the number of non-empty items in it would be x - y.